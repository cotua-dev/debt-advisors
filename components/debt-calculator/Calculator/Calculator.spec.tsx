import { render, RenderResult } from '@testing-library/react';
import { Calculator } from './index';

describe('Calculator Component', (): void => {
    let renderResult: RenderResult;

    beforeAll((): RenderResult => renderResult = render(<Calculator/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    afterAll((): void => renderResult.unmount());
});
