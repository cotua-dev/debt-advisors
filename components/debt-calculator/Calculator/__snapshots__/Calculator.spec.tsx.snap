// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component renders 1`] = `
<section
  class="debt-calculator"
>
  <h2
    class="debt-calculator-title"
  >
    Your Financial Situation
  </h2>
  <div
    class="calculator-form-wrapper"
  >
    <form
      class="calculator-form"
    >
      <div
        class="fields-wrapper"
      >
        <div
          class="field-wrapper"
        >
          <label
            class="field-label"
            for="total-unsecured-debt"
          >
            Total Unsecured Debt
            <span
              class="required"
            >
               *
            </span>
          </label>
          <input
            class="field"
            id="total-unsecured-debt"
            name="total-unsecured-debt"
            required=""
            type="number"
          />
        </div>
        <div
          class="field-wrapper"
        >
          <label
            class="field-label"
            for="months-to-pay-off"
          >
            Months To Pay Off
          </label>
          <select
            class="field"
            id="months-to-pay-off"
            name="months-to-pay-off"
          >
            <option />
            <option
              value="24"
            >
              24 months
            </option>
            <option
              value="36"
            >
              36 months
            </option>
            <option
              value="48"
            >
              48 months
            </option>
          </select>
        </div>
        <div
          class="field-wrapper"
        >
          <label
            class="field-label"
            for="average-interest-rate"
          >
            Average Interest Rate
          </label>
          <input
            class="field"
            id="average-interest-rate"
            name="average-interest-rate"
            type="number"
          />
        </div>
      </div>
      <div
        class="submit-wrapper"
      >
        <p
          class="required"
        >
          * Required
        </p>
        <button
          class="submit-button"
          type="submit"
        >
          CALCULATE
        </button>
      </div>
    </form>
  </div>
  <div
    class="calculations-table-wrapper"
  >
    <table
      class="calculations-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Credit Counseling
          </th>
          <th>
            Do Nothing (Pay Minimum)
          </th>
          <th
            class="debt-advisors"
          >
            <picture>
              <source />
              <img
                alt="Debt Advisors Regular Logo"
                class="regular-logo"
                loading="lazy"
              />
            </picture>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Total Unsecured Debt
          </td>
          <td />
          <td />
          <td
            class="debt-advisors"
          />
        </tr>
        <tr>
          <td>
            Months To Pay Off
          </td>
          <td />
          <td />
          <td
            class="debt-advisors"
          />
        </tr>
        <tr>
          <td>
            Interest Rate
          </td>
          <td />
          <td />
          <td
            class="debt-advisors"
          />
        </tr>
        <tr>
          <td>
            Extra Interest Paid
          </td>
          <td />
          <td />
          <td
            class="debt-advisors"
          />
        </tr>
        <tr>
          <td>
            Monthly Payment
          </td>
          <td />
          <td />
          <td
            class="debt-advisors"
          />
        </tr>
        <tr>
          <td>
            Your Total Cost
          </td>
          <td />
          <td />
          <td
            class="debt-advisors"
          />
        </tr>
      </tbody>
    </table>
  </div>
</section>
`;
